### PR TITLE
Use only host name as authmethod name

### LINKED ISSUE #42: AuthMethods should not contain the Policy-name
Currently, the auth method contains the name of the policy. This results in multiple auth methods for the same host, if the policy name changes (because it is based on the Puppet env for example).

However, Vault uses the first matching auth method if a client authenticates. This might be the old auth method linking to an old policy.

The cert auth method is unique and should only exist once for a given host. To ensure that, we should drop the policy name from the auth method name and override any existing auth method with the new one.

### PR DESCRIPTION
**Pull request by laugmanuel**
Fixes #42

### CODE REVIEW

File: app/models/concerns/foreman_vault/orchestration/vault_policy.rb [hunk 1]
```diff
@@ -38,11 +38,7 @@ def set_vault
         logger.info "Pushing #{name} data to Vault"
 
         vault_policy.save if vault_policy.new?
-
-        if vault_auth_method.name != old&.vault_auth_method&.name
```
Comment (laugmanuel) [inline app/models/concerns/foreman_vault/orchestration/vault_policy.rb:42]: @kamils-iRonin , `old` does not contain `vault_auth_method` and therefore this check fails every time. As a result, the existing auth method never gets updated.
Do you have a better idea, then just saving the auth method every time?

Comment (kamils-iRonin) [reply app/models/concerns/foreman_vault/orchestration/vault_policy.rb:42]: I believe that `old.vault_auth_method` should not be empty. It should return a `ForemanVault::VaultAuthMethod` object. Shouldn't we remove old/stale auth methods from Vault?

Comment (laugmanuel) [reply app/models/concerns/foreman_vault/orchestration/vault_policy.rb:42]: It is - or a least it is after Foreman 2.3 upgrade. With the above solution, we just override the auth method (if existent) or create a new one. Deletion is handled in `del_vault`.
The problem with my solution is that Foreman creates a new auth method for every host, but does not delete the existing ones (this happens once until the new authmethod exists -> it gets overridden afterwards).

**Review by kamils-iRonin — APPROVED** [review]

**Review by timogoebel — APPROVED** [review]

**Comment by timogoebel** [comment]
Thanks!

**Pull request merged**